core/src/test/java/org/elasticsearch/search/aggregations/ParsingIT.java

```diff
@@ -31,150 +31,151 @@
-                            .endObject()
-                        .endObject()
-                    .endObject()
-            .endObject()).execute().actionGet();
-    }
-
-    @Test(expected=SearchPhaseExecutionException.class)
-    public void testSameAggregationName() throws Exception {
-        createIndex("idx");
-        ensureGreen();
-        final String name = RandomStrings.randomAsciiOfLength(getRandom(), 10);
-        client().prepareSearch("idx").setAggregations(JsonXContent.contentBuilder()
-            .startObject()
-                .startObject(name)
-                    .startObject("terms")
-                        .field("field", "a")
-                    .endObject()
-                .endObject()
-                .startObject(name)
-                    .startObject("terms")
-                        .field("field", "b")
-                    .endObject()
-                .endObject()
-            .endObject()).execute().actionGet();
-    }
-
-    @Test(expected=SearchPhaseExecutionException.class)
-    public void testMissingName() throws Exception {
-        createIndex("idx");
-        ensureGreen();
-        client().prepareSearch("idx").setAggregations(JsonXContent.contentBuilder()
-            .startObject()
-                .startObject("by_date")
-                    .startObject("date_histogram")
-                        .field("field", "timestamp")
-                        .field("interval", "month")
-                    .endObject()
-                    .startObject("aggs")
-                        // the aggregation name is missing
-                        //.startObject("tag_count")
-                            .startObject("cardinality")
-                                .field("field", "tag")
-                            .endObject()
-                        //.endObject()
-                    .endObject()
-            .endObject()).execute().actionGet();
-    }
-
-    @Test(expected=SearchPhaseExecutionException.class)
-    public void testMissingType() throws Exception {
-        createIndex("idx");
-        ensureGreen();
-        client().prepareSearch("idx").setAggregations(JsonXContent.contentBuilder()
-            .startObject()
-                .startObject("by_date")
-                    .startObject("date_histogram")
-                        .field("field", "timestamp")
-                        .field("interval", "month")
-                    .endObject()
-                    .startObject("aggs")
-                        .startObject("tag_count")
-                            // the aggregation type is missing
-                            //.startObject("cardinality")
-                                .field("field", "tag")
-                            //.endObject()
-                        .endObject()
-                    .endObject()
-            .endObject()).execute().actionGet();
-    }
+    // NORELEASE move these tests to unit tests when aggs refactoring is done
```

Comment:
The problem is that because we now only allow AggregationBuilder objects to be passed into the SearchRequest these test won't compile.